Add ADR template (#39)
bf2fc6cc711aee1a0c2a/architecture: pages/about/adr-process.adoc

If, following discussion on the issue, an ADR really is called-for then open a PR to allocate an ADR id. 

This PR will add a `_adr/__nnnn__/index.adoc` file (where __nnnn__ is the ADR id) to the {{site.github.repository_name}} repo, giving your draft ADR an id. 
You can use the template in link:/adr/0/[`_adr/0/index.adoc`], so the steps to opening the PR are:

. Fork {{site.github.repository_url}}[this repo] on github
. Check out your fork (`git clone _YORKFORK_ && cd architecture`)
. Create a branch (`git checkout -b adr-__nnnn__`)
. Create your ADR file from the template (`cp -r _adr/0 _adr/__nnnn__`)
. Update your ADR file:
  * If the issue identified any applicable PADRs the `applies_padrs` property should be added to the front matter. 
  It's an array of all the applicable PADRs.

>   * If the issue identified and tags, the `tags` property should be added to the front matter. 

Reviewer: Not really sure what this is trying to say.

Author: This better?

```suggestion
  * Appropriate `tags` should be added to the front matter. 
```

It's a little vague because while people seem to like the tags idea, we're not quite sure what tags we should have, or quite how they will be used. I can imagine in a few years when there are a few hundred ADRs tags will be useful for filtering the index in order to find the decisions that affected given components, for example. Let me flip it around and ask what tags you think we should have?

Reviewer 2: Yes, better
